ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/BlockProcessor.java

```diff
@@ -168,6 +170,8 @@ void processWithdrawals(MutableBeaconState state, ExecutionPayloadSummary payloa
 
   Optional<List<Withdrawal>> getExpectedWithdrawals(BeaconState preState);
 
+  Bytes32 computeVoluntaryExitDomain(UInt64 epoch, Fork fork, Bytes32 getGenesisValidatorsRoot);
```

Comment:
I think I would change to passing the `BeaconState` and the `SignedVoluntaryExit` and do the getting the necessary stuff in the method